MobileWallet/Backup/ICloudBackup.swift

```diff
@@ -0,0 +1,342 @@
+            self?.processCloudFiles()
+        }
+    }
+
+    private func processCloudFiles() {
+        if query.resultCount == 0 { return }
+        var fileItem: NSMetadataItem?
+        var fileURL: URL?
+
+        for item in query.results {
+
+            guard let item = item as? NSMetadataItem else { continue }
+            guard let fileItemURL = item.value(forAttribute: NSMetadataItemURLKey) as? URL else { continue }
+            if fileItemURL.lastPathComponent.contains(fileName) {
+                fileItem = item
+                fileURL = fileItemURL
+            }
+        }
+
+        let fileValues = try? fileURL!.resourceValues(forKeys: [URLResourceKey.ubiquitousItemIsUploadingKey])
+        if let fileUploaded = fileItem?.value(forAttribute: NSMetadataUbiquitousItemIsUploadedKey) as? Bool, fileUploaded == true, fileValues?.ubiquitousItemIsUploading == false {
+            notifyObservers(percent: 100, completed: true, error: nil)
+            progressValue = 0.0
+            inProgress = false
+            try? cleanTempDirectory()
+        } else if let error = fileValues?.ubiquitousItemUploadingError {
+            notifyObservers(percent: 0, completed: false, error: error)
+            progressValue = 0.0
+            inProgress = false
+        } else {
+            if let fileProgress = fileItem?.value(forAttribute: NSMetadataUbiquitousItemPercentUploadedKey) as? Double {
+                notifyObservers(percent: fileProgress, completed: false, error: nil)
+                progressValue = fileProgress
+            }
+        }
+    }
+
+    private func notifyObservers(percent: Double, completed: Bool, error: Error?) {
+        observers.allObjects.forEach {
+            if let object = $0 as? ICloudBackupObserver {
+                object.didFinishUploadBackup(percent: percent, completed: completed, error: error)
+            }
+        }
+    }
+
+    private func unzipBackup(url: URL, to directory: URL) throws {
+        do {
+            try FileManager.default.unzipItem(at: url, to: directory)
+        } catch {
+            throw error
+        }
+    }
+
+    private func zipBackupFiles() throws -> URL {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd_HH-mm-ss"
+        dateFormatter.timeZone = TimeZone(secondsFromGMT: 0)
+
+        let dateString = dateFormatter.string(from: Date())
+        let archiveName = "\(fileName)_\(dateString).zip"
+
+        let tempDirectory = try tempZipDirectory()
+        let archiveURL = tempDirectory.appendingPathComponent(archiveName)
+
+        if let enumerator: FileManager.DirectoryEnumerator = FileManager.default.enumerator(atPath: directory.path) {
+            guard
+                let files = enumerator.allObjects as? [String],
+                let sqlite3File = files.first(where: { $0.hasSuffix(".sqlite3")})
```

Comment:
We know exactly the file name of the wallet database - it's a parameter passed to the CommsConfig instance. So you don't have to go through the files, you can get the file you're looking for.